Sources/Load.php

```diff
@@ -3402,28 +3347,7 @@ function cache_quick_get($key, $file, $function, $params, $level = 1)
  */
 function cache_put_data($key, $value, $ttl = 120)
 {
-	global $boardurl, $modSettings, $cache_enable, $cacheAPI;
-	global $cache_hits, $cache_count, $db_show_debug;
-
-	if (empty($cache_enable) || empty($cacheAPI))
-		return;
-
-	$cache_count = isset($cache_count) ? $cache_count + 1 : 1;
-	if (isset($db_show_debug) && $db_show_debug === true)
-	{
-		$cache_hits[$cache_count] = array('k' => $key, 'd' => 'put', 's' => $value === null ? 0 : strlen(json_encode($value)));
-		$st = microtime(true);
-	}
-
-	// The API will handle the rest.
-	$value = $value === null ? null : json_encode($value);
-	$cacheAPI->putData($key, $value, $ttl);
-
-	if (function_exists('call_integration_hook'))
-		call_integration_hook('cache_put_data', array(&$key, &$value, &$ttl));
-
-	if (isset($db_show_debug) && $db_show_debug === true)
-		$cache_hits[$cache_count]['t'] = microtime(true) - $st;
+	return StoryBB\Cache::put($key, $value, $ttl);
```

Comment:
You do a passthrough for Cache, but it looks like for Search and Mail you changed the calling code instead?

Reply:
Cache is hit in a *lot* more places than either mail or search. Additionally, caching has some instances of being called via wrapper functions.

It'll get done, just not yet because a more comprehensive review of what gets cached, when and how needs to be done. I just wanted to put the cache code somewhere sensible rather than clogging up the root of Sources...